### PR TITLE
Improved gentity struct.

### PR DESCRIPTION
**Pull request by Caball009**
**What does this PR do?**

This improves the layout of the gentity_s struct.

**How does this PR change IW4x's behaviour?**

None of the existing code should be affected. I cannot find any references to gentity fields below gentity_s::count, so there shouldn't be any offset problems.

**Anything else we should know?**

This is based on iw4 xbox360 debug information. Some of the type and field names are my own as I couldn't find their actual names. The struct change should be helpful for the antilag pr:  https://github.com/iw4x/iw4x-client/pull/169#pullrequestreview-2479676404

**Did you check all the boxes?**

- [x] Focus on a single fix or feature; remove any unrelated formatting or code changes
- [x] Mention any [related issues](https://github.com/iw4x/iw4x-client/issues) (put `closes #XXXX` in comment to auto-close issue when PR is merged)
- [x] Follow our [coding conventions](https://github.com/iw4x/iw4x-client/blob/master/CODESTYLE.md)
- [x] Minimize the number of commits

### CODE REVIEW

File: src/Game/Structs.hpp [hunk 1]
```diff
@@ -7463,6 +7463,122 @@ namespace Game
...
+		float curve;
+	};
+
+	struct missile_fields_nonGrenade
+	{
+		vec3_t curvature;
+		vec3_t targetEntOffset;
+		vec3_t targetPos;
+		vec3_t launchOrigin;
+		int stage;
```
Comment (wroyca) [inline src/Game/Structs.hpp:7537]: Type should be `MissileStage`

File: src/Game/Structs.hpp [hunk 2]
```diff
@@ -7463,6 +7463,122 @@ namespace Game
...
+		missile_fields_nonGrenade nonGrenade;
+	};
+
+	struct missile_ent_t
+	{
+		int time;
+		int timeOfBirth;
+		int travelDist;
+		vec3_t surfaceNormal;
+		int team;
```
Comment (wroyca) [inline src/Game/Structs.hpp:7552]: Type should be `team_t`

**Review by wroyca — CHANGES_REQUESTED** [review]

File: src/Game/Structs.hpp (hunk 1 above)
Comment (wroyca) [reply src/Game/Structs.hpp:7537]:
```
enum MissileStage
{
  MISSILESTAGE_SOFTLAUNCH = 0x0,
  MISSILESTAGE_ASCENT = 0x1,
  MISSILESTAGE_DESCENT = 0x2,
};
```

File: src/Game/Structs.hpp (hunk 2 above)
Comment (Caball009) [reply src/Game/Structs.hpp]: team_t already exists, but I changed the team field type to team_t.

**Review by wroyca — APPROVED** [review]

**Review by Rackover — APPROVED** [review]

File: src/Game/Structs.hpp [hunk 3]
```diff
@@ -7463,6 +7463,129 @@ namespace Game
...
+		int ammoCount;
+		int clipAmmoCount;
+		int index;
+		char dualWieldItem;
+		char padding[3];
+	};
+
+	struct spawner_ent_t
+	{
+		int team;
```
Comment (Rackover) [inline src/Game/Structs.hpp:7477]: team_t here too I suppose?

Comment (Caball009) [reply src/Game/Structs.hpp:7477]: I checked IW4, IW5 and T5 and none of them have this field as a team_t enum, but I still think it's quite likely to be a team_t.

Comment (wroyca) [reply src/Game/Structs.hpp:7477]: Can confirm, spawner_ent_t doesn't use team_t on T5; perhaps it's best to leave it as is for now and see how the code references it.

Comment (Rackover) [reply src/Game/Structs.hpp:7477]: OK; feel free to open another PR if this turns out to be team_t.  Even if that's not exactly what's in the symbols, if that's what the game does internally that it makes sense to have it be a team_t

**Pull request merged**